sdk/metrics/+opentelemetry/+sdk/+metrics/View.m

```diff
@@ -21,28 +21,48 @@
     end
 
     methods
-        function obj = View(name, description, unit, instrumentName, instrumentType, ...
-                meterName, meterVersion, meterSchemaURL, attributeKeys, ...
-                aggregation, histogramBinEdges, varargin)
+        function obj = View(options)
+            arguments
+                options.name=""
+                options.description=""
+                options.unit=""
+                options.instrumentName=""
+                options.instrumentType=""
+                options.meterName=""
+                options.meterVersion=""
+                options.meterSchemaURL=""
+                options.attributeKeys=""
+                options.aggregation=""
+                options.histogramBinEdges=[]
+            end
+            
+            instrumentTypeCategory = int32(find(options.instrumentType==["kCounter", "kHistogram", "kUpDownCounter", "kObservableCounter", "kObservableGauge", "kObservableUpDownCounter"])-1);
```

Comment:
Please use validateString

Reply:
After looking into this it looks like validateString returns a string, not an int, so I changed it to use validateString to ensure the string is valid and then find to convert it to an int. It looks like this now:

`instrument_types = ["Counter", "Histogram", "UpDownCounter", "ObservableCounter", "ObservableGauge", "ObservableUpDownCounter"];`

`instrument_type = validatestring(options.InstrumentType, instrument_types);`

`instrumentTypeCategory = int32(find(instrument_type==instrument_types)-1);`
            
Would this be a good solution?

Reply:
If you need to convert to an integer, this seems reasonable. After validatestring, at least you know find will always return an integer and not empty. Question: why do you need to convert to int32 and not just work with double?

Reply:
Since I casted the input to an int in C++ (because it represents an enum value) I needed to pass in an int32 or else there would be casting errors. I can look into alternatives if you think that isn't a good method.

Reply:
It's not a big problem, it's just in MATLAB, I tend to use doubles always and seldom use integer types. 
You could pass the double from MATLAB into C++ and cast to int in C++

Reply:
Ok sounds good, I changed it so it's a double in MATLAB and the casting is done in C++